couchbase-entity/src/main/java/com/kaufland/PostValidationException.kt

```diff
@@ -0,0 +1,17 @@
+package com.kaufland
+
+import java.lang.Exception
+import javax.lang.model.element.Element
+
+class PostValidationException : Exception {
+
+    var causingElements : List<Element> = emptyList()
+
+    constructor(message: String, vararg causingElement : Element) : super(message){
+        this.causingElements = causingElement?.toList()
```

Comment:
I think, the "?" is not necessary because causingElement is non-nullable (same for the other constructor). If you pass nothing for a vararg, it is interpreted as an empty array.